ui/src/app/views/copy-notebook-modal/component.tsx

```diff
@@ -0,0 +1,185 @@
+import * as React from 'react';
+
+import { Button } from 'app/components/buttons';
+import { styles as headerStyles } from 'app/components/headers';
+import { Select, TextInput } from 'app/components/inputs';
+import { Modal, ModalBody, ModalFooter, ModalTitle } from 'app/components/modals';
+import { FileDetail, Workspace } from 'generated/fetch';
+
+import { Spinner } from 'app/components/spinners';
+import { workspacesApi } from 'app/services/swagger-fetch-clients';
+import { WorkspacePermissions } from 'app/utils/workspace-permissions';
+
+enum State { FORM, ERROR, SUCCESS }
+
+export interface CopyNotebookModalProps {
+  fromWorkspaceNamespace: string;
+  fromWorkspaceName: string;
+  fromNotebook: FileDetail;
+  onClose: Function;
+  onCopy: Function;
+}
+
+export interface CopyNotebookModalState {
+  writeableWorkspaces: Array<Workspace>;
+  destination: Workspace;
+  newName: string;
+  state: State;
+  errorMsg: string;
+  loading: boolean;
+}
+
+const boldStyle = {
+  fontWeight: 700
+};
+
+export class CopyNotebookModal extends React.Component<CopyNotebookModalProps,
+CopyNotebookModalState> {
+  constructor(props: CopyNotebookModalProps) {
+    super(props);
+
+    this.state = {
+      writeableWorkspaces: [],
+      newName: props.fromNotebook.name,
+      destination: null,
+      state: State.FORM,
+      errorMsg: '',
+      loading: true
+    };
+  }
+
+  componentDidMount() {
+    workspacesApi().getWorkspaces()
+      .then((response) => {
+        this.setState({
+          writeableWorkspaces: response.items
+            .filter(item => new WorkspacePermissions(item).canWrite)
+            .map(workspaceResponse => workspaceResponse.workspace),
+          loading: false
+        });
+      });
+  }
+
+  save() {
+    this.setState({ loading: true });
+
+    workspacesApi().copyNotebook(
+      this.props.fromWorkspaceNamespace,
+      this.props.fromWorkspaceName,
+      this.props.fromNotebook.name,
+      {
+        toWorkspaceName: this.state.destination.id,
+        toWorkspaceNamespace: this.state.destination.namespace,
+        newName: this.state.newName
+      }
+      ).then((response) => {
+        this.setState({ state: State.SUCCESS, loading: false });
```

Comment:
I followed the Terra UI for this. I agree that it would simplify the code but I think its a nice UI touch to be able to decide where to go after. Ex. what if you want to copy a bunch of notebooks?